addons/website_crm/models/website_visitor.py

```diff
@@ -7,12 +7,19 @@
 class WebsiteVisitor(models.Model):
     _inherit = 'website.visitor'
 
-    lead_ids = fields.One2many('crm.lead', 'visitor_id', string='Leads')
+    lead_ids = fields.Many2many('crm.lead', string='Leads')
```

Comment:
I have some questions here ...Visitor is a marketing oriented record. It is not obvious people have rights on leads. Previously computation of lead count would either crash, either give results depending on ACLS. Currently you always have results independently of your access rights.

I would
 * display lead_ids and lead_count only if people are at least sales something group as in all other modules;
 * remove the hand made sql and replace it by a real search depending on acls;
 * add some rights on visitor model in website_crm for lead people as it does not seem to be the case;